doc/src/modules/polys/numberfields.rst

```diff
@@ -0,0 +1,289 @@
+real line.
+
+.. autofunction:: isolate
+
+
+As elements of finitely-generated modules
+`````````````````````````````````````````
+
+In computational algebraic number theory, finitely-generated
+$\mathbb{Z}$-modules are of central importance. For example, every
+order_ and every ideal_ is such a module.
+
+In particular, the maximal order -- or `ring of integers`_ -- in a number field
+is a finitely-generated $\mathbb{Z}$-module, whose generators form an
+`integral basis`_ for the field.
+
+Classes allowing us to represent such modules, and their elements, are provided
+in the :py:mod:`~.modules` module. Here, the :py:class:`~.ModuleElement` class
+provides another way to represent algebraic numbers.
+
+
+Finitely-generated modules
+--------------------------
+
+.. automodule:: sympy.polys.numberfields.modules
+
+Class Reference
+```````````````
+.. autoclass:: Module
+   :members:
+
+   .. automethod:: Module.__call__
+
+.. autoclass:: PowerBasis
+   :members:
+
+   .. automethod:: PowerBasis.__init__
+
+.. autoclass:: Submodule
+   :members:
+
+   .. automethod:: Submodule.__init__
+
+.. autoclass:: ModuleElement
+   :members:
+
+   .. automethod:: ModuleElement.__init__
+   .. automethod:: ModuleElement.__add__
+   .. automethod:: ModuleElement.__mul__
+   .. automethod:: ModuleElement.__mod__
+
+.. autoclass:: PowerBasisElement
+   :members:
+
+.. autofunction:: make_mod_elt
+
+.. autoclass:: ModuleHomomorphism
+   :members:
+
+   .. automethod:: ModuleHomomorphism.__init__
+
+.. autoclass:: ModuleEndomorphism
+   :members:
+
+   .. automethod:: ModuleEndomorphism.__init__
+
+.. autoclass:: InnerEndomorphism
+   :members:
+
+   .. automethod:: InnerEndomorphism.__init__
+
+.. autoclass:: EndomorphismRing
+   :members:
+
+   .. automethod:: EndomorphismRing.__init__
+
+.. autofunction:: find_min_poly
+
+
+Utilities
+---------
+
+.. currentmodule:: sympy.polys.numberfields.utilities
+
+.. autofunction:: is_rat
+.. autofunction:: is_int
+.. autofunction:: get_num_denom
+.. autofunction:: extract_fundamental_discriminant
+
+.. autoclass:: AlgIntPowers
+   :members:
+
+   .. automethod:: AlgIntPowers.__init__
+
+.. autofunction:: coeff_search
+.. autofunction:: supplement_a_subspace
+
+
+Solving the Main Problems
+=========================
+
+Integral Basis
+--------------
+
+.. currentmodule:: sympy.polys.numberfields
+.. autofunction:: round_two
+
+
+Prime Decomposition
+-------------------
+
+.. autofunction:: prime_decomp
+.. currentmodule:: sympy.polys.numberfields.primes
+.. autoclass:: PrimeIdeal
+   :members:
+
+   .. automethod:: PrimeIdeal.__init__
+
+
+p-adic Valuation
+----------------
+
+.. currentmodule:: sympy.polys.numberfields
+.. autofunction:: prime_valuation
+
+
+Finding Minimal Polynomials
+---------------------------
+
+.. autofunction:: minimal_polynomial
+.. autofunction:: minpoly
+
+
+The Primitive Element Problem
+-----------------------------
+
+.. autofunction:: primitive_element
```

Comment:
This function is badly under-documented. When I wrote the other docs for domains this one slipped under the radar.